### ISSUE TITLE
Bad CLI practice ! yarn with no argument must no modify anything on my computer

### ISSUE THREAD

**Issue by tmiele** [yarnpkg/yarn#7620]
## Current Behavior
```bash
C:\Users\Thomas
λ yarn
yarn install v1.19.1
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...

success Saved lockfile.
Done in 0.07s.

C:\Users\Thomas
λ ls yarn.lock
yarn.lock
```
After this i get a yarn.lock new file in my user directory.

## What I expected
When i execute yarn just after fresh install of yarn on Windows 10 home edition 64 bit in Cmder I expect no --help or --version but a compact and reduce help about

## Env
```bash
C:\Users\Thomas
λ node --version
v12.11.1

C:\Users\Thomas
λ npm --version
6.11.3

C:\Users\Thomas
λ yarn --version
1.19.1
```

**Comment by burtek** [comment]
I actually expect it to be shorthand of `yarn install`, as it always was 🤷‍♂  It is well documented:

>Running `yarn` with no command will run `yarn install`, passing through any provided flags.

https://yarnpkg.com/en/docs/cli/install

**Comment by jacobq** [comment]
CLI convention (and hence what constitutes "bad CLI practice") varies by OS, and `yarn` is a multi-platform tool. Hence more objective goals are that it be consistent, ergonomic, and well-documented. Admittedly, it has different semantics than `npm`, but these are intentional are arguably more ergonomic (e.g. optimized for most common use cases).

Besides, even in Windows' `cmd.exe` there are commands like `START` or `EXIT` which take action without any arguments. You wouldn't expect `dir` (or `ls`) to print help information when run without arguments, right? Of course not. Instead, it's optimized for the very common use case of listing the contents of the current directory (rather than having to explicitly specific this with a `.` argument). `yarn` is similar. By far the most common action needed is installing packages listed in `package.json` / `yarn.lock`, so that is what it does when run without arguments.

**Comment by BYK** [comment]
This is working as intended as mentioned earlier. Although we appreciate the feedback, we don't have any plans for changing this behavior which would also require a major version bump as it would be a breaking change.

Closing as this is not really actionable.

**Comment by DanielGoldfarb** [comment]
I realize that this behavior of **`yarn`** is *not* going to change, and as I am currently learning `yarn` I will also learn to get used to it.  

That said, my two cents fwiw: the most widely accepted "best practices" in both CLI (and GUI) **is** ***not to modify the machine*** (the files, installs, config, etc.) in any way without arguments (or without confirmation in the case of GUI).  

*All* of the [above examples](https://github.com/yarnpkg/yarn/issues/7620#issuecomment-564685296), of commands that act without arguments, are commands that *do **not** modify the machine.*   

There may be *some* exceptions (commands that *do* modify the machine), but these exceptions are small minority, and typically are *single purpose commands* such as **`rm`** (whose sole purpose is to remove files, and even `rm` ultimately requires *some* kind of argument).  Can't think of other exceptions off the top of my head right now; I sure there may be some but I'm also sure they are very few.  

In simple terms, I plainly *disagree* that "`what constitutes "bad CLI practice") varies by OS`" ... I do believe that it *is generally accepted best practice not to modify the machine without arguments or confirmation of some sort*.  

For historical reasons, I would not expect `yarn` to change (but nice if it did).  At least, it would be nice if there was a way to configure `yarn` *not* to default to installing packages.

**Comment by russeg** [comment]
Annoying and pretty dumb imo. I only want yarn do to something when it finds a  package.json so i have this:

```bash
yarn() {
    if [ $# -eq 0 ] && [ ! -f package.json ]; then
        command yarn --help
    else
        command yarn "$@"
    fi
}
